src/app/models/user.rb

```diff
@@ -11,4 +11,15 @@ def update_credit(amount)
   def set_default_credit
     self.credit = DEFAULT_CREDIT
   end
-end
\ No newline at end of file
+
+  def self.from_omniauth(auth)
+    User.where(provider: auth.provider, uid: auth.uid).first_or_initialize do |user|
+      user.provider = auth.provider
+      user.uid = auth.uid
+      user.name = auth.info.name
+      user.oauth_token = auth.credentials.token
+      user.oauth_expires_at = Time.at(auth.credentials.expires_at)
+      user.save!
```

Comment:
Perhaps it might be better to `save` and redirect to the schedule page on success and back to the homepage if there was an error.